### DISCUSSION TITLE
Custom font cannot be found when in a Styles file

### DISCUSSION THREAD

**Discussion by itslyn** [AvaloniaUI/Avalonia#13785] [Q&A] [answered]
I want to use a custom font in my program. I currently have the font files included in my project as shown here, and they are all set to have build action as AvaloniaResource.
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/ea36f2be-f4a3-4913-bd2d-d629d91f1b8c)

I have also included it in my App.axaml file:
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/bcb22572-d8ce-409a-8267-a2f8c612a638)

I currently have a style for my TextBlock in a view's Grid.Styles here, and it _does_ work and successfully styles my TextBlocks with the custom font.
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/481b2af6-25c7-49a4-a7c0-32827a110f23)

However, I want to move this style to a separate axaml file that has all my styles in them so that I can use them across my program without having to have it in every view. I have created an axaml file for my style here and moved the style code from my UserControl to this file. However, it _does not_ work and when I try to run my program, I get an error that says that the static resource can't be found.
![image](https://github.com/AvaloniaUI/Avalonia/assets/115854752/bbde92c6-9f34-4a89-bdfc-5eeab4edf57b)

How come the static resource for my custom font is able to be found when being used in my UserControl, but cannot be found when it is in a styles file? Is it because the Styles file is being run before the App.axaml file? But that doesn't make sense to me because my Styles file has the same build action as my UserControl. Am I missing some code to make it work? TIA

**Comment by IanRawley** [comment, marked as answer]
Swap the order of the Styles and Resources in App.axaml  
The parser is loading them in order, and while it's processing the styles in the file the resources in App.axaml haven't been processed yet.  Ergo they don't exist and that's an error.
